sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveLateralColumnAliasReference.scala

```diff
@@ -168,19 +168,18 @@ object ResolveLateralColumnAliasReference extends Rule[LogicalPlan] {
             && aggregateExpressions.exists(_.containsPattern(LATERAL_COLUMN_ALIAS_REFERENCE)) =>
 
           // Check if current Aggregate is eligible to lift up with Project: the aggregate
-          // expression only contains: 1) aggregate functions, 2) grouping expressions, 3) lateral
-          // column alias reference or 4) literals.
+          // expression only contains: 1) aggregate functions, 2) grouping expressions, 3) leaf
+          // expressions excluding attributes not in grouping expressions
```

Comment:
what's the actual change? do we allow something new?

Reply:
Check the tests: `Leaf expression as aggregate expressions should be eligible to lift up`.

Reply:
tldr, previously it doesn't allow leaf expressions other than Literal and LateralColumnAliasReference, now it allows.